mage/query-modules/python/temporal-graph-networks.md

```diff
@@ -0,0 +1,415 @@
+  * `device_type: str`  ➡ **[OPTIONAL]** type of device you want to use for training - cuda or cpu
+  * `node_features_property: str`➡ **[OPTIONAL]** name of features property on nodes from which we read features
+  * `edge_features_property: str`➡ **[OPTIONAL]** name of features property on edges from which we read features
+  * `node_label_property: str` ➡ **[OPTIONAL]** name of label property on nodes from which we read features
+#### Usage:
+
+```cypher
+ CALL tgn.set_params({learning_type:'self_supervised', batch_size:200, num_of_layers:2, 
+                      layer_type:'graph_attn',memory_dimension:20, time_dimension:50, 
+                      num_edge_features:20, num_node_features:20, message_dimension:100, 
+                      num_neighbors:15, edge_message_function_type:'identity',
+                      message_aggregator_type:'last', memory_updater_type:'gru', num_attention_heads:1});
+```
+
+
+### `update(edges)`
+
+#### Input:
+
+* `edges: mgp.List[mgp.Edges]` ➡ list of edges that arrive in a stream to `Memgraph` database to preprocess. If a batch is full, `train` or `eval` starts, depending on the mode.
+
+#### Usage:
+
+There are a few options here:
+
+The most convenient one is to create a **[trigger](https://memgraph.com/docs/memgraph/reference-guide/triggers)** so that every
+time an edge is added to the graph, the trigger calls the procedure and makes an
+update.
+
+```cypher
+CREATE TRIGGER create_embeddings ON --> CREATE BEFORE COMMIT
+EXECUTE CALL tgn.update(createdEdges) RETURN 1;
+```
+
+The second option is to add all the edges and then call the algorithm with them:
+
+```cypher
+MATCH (n)-[e]->(m)
+WITH COLLECT(e) as edges
+CALL tgn.update(edges) RETURN 1;
+```
+
+
+### `get()`
+
+#### Output:
+
+* `node: mgp.Vertex` ➡ vertex in Memgraph database
+* `embedding: mgp.List[float]` ➡ low-dimensional representation of node in form of graph embedding
+
+#### Usage:
+
+```cypher
+CALL tgn.get() YIELD * RETURN *;
+```
+
+
+### `set_eval()`
+
+With this function, you change the mode of **temporal graph networks** to `eval` mode. Any new edges which arrive
+will **not** be used to **train** the module, but to **evaluate**.
+
+If you know when the stream will end, it is good to change the mode to "eval" and test how your model performs. 
+Once the stream finishes, a good option is to run a few training epochs to get the best results.
+
+
+#### Usage:
+
+```cypher
+CALL tgn.set_eval() YIELD *;
+```
+
+
+### `get_results()`
+
+#### Output:
+
+* `epoch_num=mgp.Number` ➡ number of `train` or `eval` epochs
+* `batch_num=mgp.Number` ➡ number of batches per `train` or `eval` epoch
+* `batch_process_time=mgp.Number` ➡ time needed to process batch 
+* `accuracy=mgp.Number` ➡ mean average precision on current batch
+* `accuracy_type=str` ➡ type of MAP performed, "train" or "eval"
+
+
+#### Usage:
+
+```cypher
+CALL tgn.get_results() YIELD * RETURN *;
+```
+
+
+### `train_and_eval(num_epochs)`
```

Comment:
I would love to have this explained in the above section. It's more useful than having so long section of implementation details. Having both is fine too